packages/form-js-viewer/src/core/ConditionChecker.js

```diff
@@ -0,0 +1,79 @@
+import { unaryTest } from 'feelin';
+import { isString } from 'min-dash';
+
+export class ConditionChecker {
+  constructor(formFieldRegistry) {
+    this._formFieldRegistry = formFieldRegistry;
+  }
+
+  /**
+   * For given data, remove properties for which conditions are not met.
+   *
+   * @param {Object<string, any>} properties
+   * @param {Object<string, any>} data
+   */
+  applyConditions(properties, data = {}) {
+    const conditions = this._getConditions();
+
+    const newProperties = { ...properties };
+    for (const { key, condition } of conditions) {
+      const shouldRemove = !this.check(condition, data);
+
+      if (shouldRemove) {
+        delete newProperties[ key ];
+      }
+    }
+
+    return newProperties;
+  }
+
+  /**
+   * Check if given condition is met.
+   *
+   * @param {string} condition
+   * @param {import('../types').Data} [data]
+   *
+   * @returns {boolean}
+   */
+  check(condition, data = {}) {
+    if (!condition) {
+      return true;
+    }
+
+    if (!isString(condition) || !condition.startsWith('=')) {
+      return false;
+    }
+
+    try {
+      const result = unaryTest(condition.slice(1), data);
+
+      return result;
+    } catch (error) {
+      return false;
```

Comment:
Consider not swallowing the error, but logging it (or emitting it via `eventBus.fire(`error`, ...)`.

Reply:
Implemented via https://github.com/bpmn-io/form-js/pull/403/commits/848b703812998668b302397f73c8b416abbb094a